optimera-htb.js

```diff
@@ -265,139 +285,143 @@ function OptimeraHtb(configs) {
-            //? }
-
-            var pubKitAdId = RenderService.registerAd({
-                sessionId: sessionId,
-                partnerId: __profile.partnerId,
-                adm: bidCreative,
-                requestId: curReturnParcel.requestId,
-                size: curReturnParcel.size,
-                price: targetingCpm,
-                dealId: bidDealId || undefined,
-                timeOfExpiry: __profile.features.demandExpiry.enabled ? (__profile.features.demandExpiry.value + System.now()) : 0,
-                auxFn: __renderPixel,
-                auxArgs: [pixelUrl]
-            });
+          for (var j = 0; j < returnParcels.length; j++) {
 
-            //? if (FEATURES.INTERNAL_RENDER) {
-            curReturnParcel.targeting.pubKitAdId = pubKitAdId;
-            //? }
+              var curReturnParcel = returnParcels[j];
+
+              var headerStatsInfo = {};
+              var htSlotId = curReturnParcel.htSlot.getId();
+              headerStatsInfo[htSlotId] = {};
+              headerStatsInfo[htSlotId][curReturnParcel.requestId] = [curReturnParcel.xSlotName];
+
+              var curBid;
+
+              for (var i = 0; i < bids.length; i++) {
+
+                  /**
+                   * This section maps internal returnParcels and demand returned from the bid request.
+                   * In order to match them correctly, they must be matched via some criteria. This
+                   * is usually some sort of placements or inventory codes. Please replace the someCriteria
+                   * key to a key that represents the placement in the configuration and in the bid responses.
+                   */
+
+                  /* ----------- Fill this out to find a matching bid for the current parcel ------------- */
+
+                  var divID = curReturnParcel.xSlotRef.divID;
+                  var bidObj = bids[i];
+                  var bidObjProps = Object.getOwnPropertyNames(bidObj);
+
+                  if ( bidObjProps.includes(curReturnParcel.xSlotRef.divID) ) {
+                      curBid = bids[i];
+                      bids.splice(i, 1);
+                      break;
+                  }
+              }
+
+              /* No matching bid found so its a pass */
+              if (!curBid) {
+                  if (__profile.enabledAnalytics.requestTime) {
+                      __baseClass._emitStatsEvent(sessionId, 'hs_slot_pass', headerStatsInfo);
+                  }
+                  curReturnParcel.pass = true;
+                  continue;
+              }
+
+              /* ---------- Fill the bid variables with data from the bid response here. ------------*/
+
+              /* Using the above variable, curBid, extract various information about the bid and assign it to
+               * these local variables */
+
+              /* the bid price for the given slot */
+              var bidPrice = 1;
```

Comment:
Hi @ix-certification 

Thank you for these comments. Keith has responded more comprehensively in the email thread, but just logging one of the responses here as well.

We are not providing a creative, a price, or anything beyond a need to push the variables into DFP. You can think of the functionality more like a DMP pushing in an audience ID rather than an auction result.